### ISSUE TITLE
Question on CrossEntropyMetric in loss.py line 159 .

### ISSUE THREAD

**Issue by kobenaxie** [awslabs/sockeye#200]
when computing smoothed cross entropy , my understand is the function of this line:
`label_dist = mx.nd.where(ignore, label_dist, mx.nd.zeros_like(label_dist))`
is to set the distribution of PAD to be zero, while keeping the distribution None-PAD elements in label after label smoothed,  if so , maybe the condition in nd.where() should be `1-ignore` rather than `ignore`, is not it ?

**Comment by fhieber** [comment]
You are absolutely right, thanks!
This should be `1-ignore` (or `label`). Looking at the git history, we must have added this bug with the mxnet0.12 update.
This should only affect cases where you train models with label smoothing and the eve optimizer (which uses these loss classes to get access to the loss function value).

Do you want to open a Pull Request with a fix?

**Comment by kobenaxie** [comment]
I just wanna confirm my understanding is correct, and have fixed it.
